/*
 * This file was automatically generated by EvoSuite
 */

package org.prevayler.foundation;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.evosuite.junit.EvoSuiteRunner;
import static org.junit.Assert.*;
import org.prevayler.foundation.Cool;

@RunWith(EvoSuiteRunner.class)
public class TestCool_2 {


  //Test case number: 2
  /*
   * 3 covered goals:
   * 1 Weak Mutation 5: org.prevayler.foundation.Cool.unexpected(Ljava/lang/Exception;)V:23 - ReplaceConstant - Unexpected Exception was thrown. -> 
   * 2 org.prevayler.foundation.Cool.unexpected(Ljava/lang/Exception;)V: root-Branch
   * 3 Weak Mutation 5: org.prevayler.foundation.Cool.unexpected(Ljava/lang/Exception;)V:23 - ReplaceConstant - Unexpected Exception was thrown. -> 
   */
  @Test
  public void test2()  throws Throwable  {
      // Undeclared exception!
      try {
        Cool.unexpected((Exception) null);
        fail("Expecting exception: RuntimeException");
      } catch(RuntimeException e) {
        /*
         * Unexpected Exception was thrown.
         */
      }
  }
}
